contracts/Uni.sol

```diff
@@ -110,6 +110,23 @@ contract Uni {
         _moveDelegates(address(0), delegates[dst], amount);
     }
 
+    /**
+     * @notice Burns tokens
+     * @param rawAmount The number of tokens to be minted
+     */
+    function burn(uint rawAmount) external {
+        uint96 amount = safe96(rawAmount, "Vexchange::approve: amount exceeds 96 bits");
+        require(amount <= balances[msg.sender], "Vexchange::burn token balance is less than the burn amount");
+
+        balances[msg.sender] = sub96(balances[msg.sender], amount, "Vexchange::burn new balance underflows");
```

Comment:
Given we are checking underflow here, is the `require(amount <= balances[msg.sender])` unnecessary?

Reply:
Yes this is something I've considered when I was writing it. Logically there is no way that either line 119 or line 121 would be valid without the other being invalid. I guess I wanted to be really safe when I was writing it. I will delete line 119 in this case